packages/nodes-base/types.d.ts

```diff
@@ -0,0 +1,24 @@
+import type { FtpCredentialSchema } from './credentials/Ftp.credentials';
+import type { SftpCredentialSchema } from './credentials/Sftp.credentials';
+import type { StrapiApiCredential } from './credentials/StrapiApi.credentials';
+import type { StrapiTokenApiCredential } from './credentials/StrapiTokenApi.credentials';
+
+type CredentialSchemaMap = {
+	strapiApi: StrapiApiCredential;
+	strapiTokenApi: StrapiTokenApiCredential;
+	ftp: FtpCredentialSchema;
+	sftp: SftpCredentialSchema;
+};
+
+declare module 'n8n-workflow' {
+	interface FunctionsBase {
+		getCredentials<Type extends keyof CredentialSchemaMap | {}>(
```

Comment:
I like the idea of `getCredentials` automatically inferring the return type based on the `type` string, but having to maintain this additional map for 360+ credentials is going to be quite a chore. 
Maybe we should update `getCredentials` to alternatively accept the credential class as the type argument, and infer the return type from that instead somehow?

Reply:
Good Idea 😄 implemented!